### ISSUE TITLE
Many samples don't work for me

### ISSUE THREAD

**Issue by KarelPeeters** [NVIDIA/cudnn-frontend#30]
I'm familiar with how to use CUDA and more specifically cuDNN., and I'm now trying to get started with the cuDNN backend API by experimenting with the samples in this repository. However, many of them don't seem to work for me.

### My environment

* Windows 11 version 21H2 (OS Build 22000.613)
* Cuda 1.5 V11.5.119
* Cudnn 8.3.3
* Visual Studio 2019 version 16.0
* (the machine I'm compiling this on only has a GTX 1060 MaxQ, but I don't think that's very relevant yet)

### General errors

* `ConvBiasAct`: Setting a relu clip slope doesn't work, the result is always 
`[ERROR] Exception CUDNN_BACKEND_POINTWISE_DESCRIPTOR: SetAttribute CUDNN_ATTR_POINTWISE_RELU_LOWER_CLIP_SLOPE, Failed cudnn_status: CUDNN_STATUS_BAD_PARAM`

* `ConvBiasAct`: After commenting out the previous code, the next issue is 
`[ERROR] Exception CUDNN_BACKEND_EXECUTION_PLAN_DESCRIPTOR: GetAttribute CUDNN_ATTR_ENGINE_BEHAVIOR_NOTE Failed cudnn_status: CUDNN_STATUS_BAD_PARAM`.
This code is gated behind a cuDNN version check, but my version is high enough so I don't know why this fails. After commenting out this and the previous issues this sample is running fine.

* `ConvBiasScaleAct`, `ConvBiasScaleAct_int8`, `ConvScaleBiasAddAct sample`, `ConvScaleBiasAddAct sample_float` and fail with the exact same
`[ERROR] Exception CUDNN_BACKEND_ENGINEHEUR_DESCRIPTOR: cudnn Finalize failed cudnn_status: CUDNN_STATUS_BAD_PARAM`.
No idea what could cause this.

* `Multihead attention sample`, `MatmulBiasAct sample` and `MatmulBiasAct sample_float`, all fail with
`[ERROR] Exception CUDNN_BACKEND_MATMUL_DESCRIPTOR: cudnnCreate Failed cudnn_status: CUDNN_STATUS_ALLOC_FAILED`.
I'm not sure what could cause this, I have plenty of free CPU and GPU memory.

* `ConvDrelu` and `DgradDrelu` fail with
`[ERROR] Exception CUDNN_BACKEND_OPERATION: SetAttribute CUDNN_ATTR_OPERATION_POINTWISE_DYDESC Failed cudnn_status: CUDNN_STATUS_BAD_PARAM`
Maybe pointwise derivatives are properly supported yet? There is no documentation in the API reference for this attribute name yet.

### Hard-crash in `BN Finalize`:

* Creation of the third tensor here fails: https://github.com/NVIDIA/cudnn-frontend/blob/43709ab96c47e26eebcdac72f93f946d44ceffa8/samples/fusion_sample.cpp#L2370-L2372. I assume this is because there is some issue with the `CUDNN_DATA_INT64` datatype? Specifically the `cudnnBackendFinalize` of the tensor descriptor crashes internally (it's not just a non-success return value, the call itself actually generates an exception).

### Conclusion

In conclusion, everything seems to be very brittle at the moment. Is this because I am using some wrong versions? Or is the cuDNN backend/frontend API still too much of a work-in-progress for end users? Or is Windows support still lacking?

**Comment by Anerudhan** [comment]
Hi @KarelPeeters,
Sorry for the delayed responses. We have been trying to improve the stability of cuDNN FE with every cuDNN release.
We have the following features in immediate roadmap.
- Better Frontend tests in general.
- Better Windows testing and coverage.

I hope with the Frontend becoming more mature, most of these issues would have been resolved. 

Can you please give it a try with v0.7 and cuDNN 8.5.0 ?

Let us know if your issues persist, we will track internally and get back to you.

Thanks, 
Anerudhan

**Comment by KarelPeeters** [comment]
There are 10 tests that immediately cause fatal errors and crash the entire test runner, causes are:
* access violations
* calling front() on an empty vector
* division by zero (not 100% sure about this one)

The rest of the tests aren't that successful either:
![image](https://user-images.githubusercontent.com/7887115/193290302-3d71825e-3a16-48e3-b395-9a44f52beb2e.png)

The errors seem to still be mostly the same as before.

As a side note, to even build everything requires me to edit `CMakeLists.txt`:
* remove `/WX` (which treats warnings as errors), since MSVC emits a bunch of warnings on the current code.
* change `link_directories($ENV{CUDA_PATH}/lib64)` to `link_directories($ENV{CUDA_PATH}/lib/x64)` which is where CUDA is actually installed by the installer
* remove `target_link_libraries(Samples m)`, MSVC does not have a separate math library to link.

This is all with CUDA `v11.7` and CUDNN `v8.5.0`.

**Comment by Anerudhan** [comment]
Thanks for reporting this. 

- We are working on better MSVC support on next frontend release.

Aside,
Can you please let us know which GPU are you running this on ?

Thanks

**Comment by KarelPeeters** [comment]
Still the same (old and probably unsupported)  GTX 1060 MaxQ. This issue was not really about requesting full support for everything, just some of the more severe issues using this on Windows has.

**Comment by mnicely** [comment]
We have refreshed our samples to support the newest frontend. You can find new examples [here](https://github.com/NVIDIA/cudnn-frontend/tree/main/samples). Closing this for now.
